src/mono/mono/mini/interp/interp.c

```diff
@@ -3703,6 +3709,11 @@ interp_exec_method (InterpFrame *frame, ThreadContext *context, FrameClauseArgs
 			 * than the callee stack frame (at the interp level)
 			 */
 			context->stack_pointer = (guchar*)frame->stack + new_method->alloca_size;
+			if (G_UNLIKELY (context->stack_pointer >= context->stack_end)) {
+				context->stack_end = context->stack_real_end;
```

Comment:
I don't think we should be changing `stack_end`.

Reply:
Without it, the other asserts will be hit during handling the exception.

Reply:
Maybe we could change the `stack_end` here, in order to allow some more space for filters and finally to run if they need more stack for whatever reason, but set back the `stack_end` to the original value in order to preserve functionality for multiple stack overflows. (I think we can reset it in `interp_set_resume_state`). In that case, asserting for the real end might make more sense.

Reply:
Going to merge this for now since it might help diagnose
https://github.com/dotnet/runtime/issues/65335